pkg/contextstore/store.go

```diff
@@ -0,0 +1,290 @@
+package contextstore
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const configFileName = "config.json"
+
+// Store provides a context store for easily remembering endpoints configuration
+type Store interface {
+	GetCurrentContext() string
+	SetCurrentContext(name string) error
+	ListContexts() (map[string]ContextMetadata, error)
+	CreateOrUpdateContext(name string, meta ContextMetadata) error
+	GetContextMetadata(name string) (ContextMetadata, error)
+	ResetContextTLSMaterial(name string, data *ContextTLSData) error
```

Comment:
Are these 2 methods `ResetContextTLSMaterial` and `ResetContextEndpointTLSMaterial` really needed on this interface?
I feel like you can do all this using `CreateOrUpdateContext`

Reply:
This would not be so practical:
- if you want to add a new endpoint to an existing context, you don't want to read/write back TLS data for all endpoints
- IMO the ResetContextTLSMaterial might not be needed, but it does not cost so much to have it